client/lib/domains/index.js

```diff
@@ -87,6 +87,10 @@ function canRedirect( siteId, domainName, onComplete ) {
 		return;
 	}
 
+	if ( ! domainName.match(/^(http)/) ) {
```

Comment:
@klimeryk Fixed it. Let me know thoughts.

I think I'll look at https:// in another PR, although not sure what the case is. Are you talking about https:// in the redirect or prepending https:// ?
I entered https://example.com/and and it let me go to the next step.